internal/users/db/migration.go

```diff
@@ -273,7 +274,14 @@ func renameUsersInGroupFile(oldNames, newNames []string) error {
 		return fmt.Errorf("error writing %s: %w", tempPath, err)
 	}
 
-	if err := os.Rename(tempPath, groupFile); err != nil {
+	err = fileutils.Lrename(tempPath, groupFile)
+	var symlinkErr *fileutils.SymlinkResolutionError
+	if errors.As(err, &symlinkErr) {
+		log.Warning(context.Background(), err.Error())
+		// If the symlink resolution fails, we rename the symlink without resolving it.
+		err = os.Rename(tempPath, groupFile)
```

Comment:
With current code is quite hard (if not impossible) to reach this since the initial `content, err := os.ReadFile(groupFile)` would not work if the group file points to something that cannot be resolved.

Reply:
Well if we have te lock it shouldn't, no?

And if it does, then there's a system issue.

Reply:
I think there's always a system issue if `/etc/groups` becomes a dangling symlink.

Reply:
Yeah, indeed... I mean, if we assume that it can happen, then we're broken way earlier than that... So, IMHO or we accept "fixing it", or we can't be bothered and we do nothing special here, as it goes a bit of out of scope of our groups management duties.

Reply:
> or we can't be bothered and we do nothing special here, as it goes a bit of out of scope of our groups management duties

fine by me (see https://github.com/ubuntu/authd/pull/723#discussion_r2063252916)